code/addons/interactions/template/stories/basics.stories.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable jest/no-standalone-expect */
+import globalThis from 'global';
+import { within, waitFor, fireEvent, userEvent } from '@storybook/testing-library';
+import { expect } from '@storybook/jest';
+
+export default {
+  component: globalThis.Components.Form,
+  argTypes: {
+    onSuccess: { type: 'function' },
+  },
+};
+
+export const Type = {
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+    await userEvent.type(canvas.getByTestId('value'), 'test');
+  },
+};
+
+export const Step = {
+  play: async (context) => {
+    await context.step('Enter value', async () => Type.play(context));
+  },
+};
+
+export const Click = {
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+    await fireEvent.click(canvas.getByTestId('succeed'));
+  },
+};
+
+export const Callback = {
+  play: async (context) => {
+    const { args, canvasElement, step } = context;
+    const canvas = within(canvasElement);
+    await step('Enter value', async () => Type.play(context));
+
+    await step('Click checkbox', async () => Click.play(context));
+
+    await step('Submit', async () => {
+      await fireEvent.click(canvas.getByRole('button'));
+    });
+
+    await expect(args.onSuccess).toHaveBeenCalled();
+  },
+};
+
+export const WaitFor = {
+  play: async (context) => {
+    const { args, canvasElement, step } = context;
+    const canvas = within(canvasElement);
+    await step('Enter value', async () => Type.play(context));
+
+    await step('Submit', async () => {
+      await fireEvent.click(canvas.getByRole('button'));
+    });
+
+    await expect(args.onSuccess).not.toHaveBeenCalled();
+
+    await waitFor(
+      async function () {
+        await canvas.getByTestId("Submitted 'test' when not allowed!");
+      },
+      { timeout: 2000 }
+    );
+  },
+};
+
+export const Hover = {
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+    await userEvent.hover(canvas.getByRole('button'));
```

Comment:
Perhaps we should note here that `userEvent.hover` will not trigger `:hover` pseudo styles, but rather simulates a `mouseenter` event.

I've considered tweaking these userEvent methods to actually call out to the pseudo states addon if it's installed, but perhaps we should simply provide a direct `pseudoState.hover` interaction for people to use.

The pseudo states addon won't trigger default browser styles by the way, so it wouldn't have any effect on your unstyled demo form.

Reply:
So is `userEvent.hover()` only useful for testing JS-based hover interactions? If so I'll probably drop this test as the cost benefit of implementing the interaction in each render isn't worth it.

Reply:
Got rid of it, no point in testing `hover` if it requires JS code in our form. We aren't testing testing-lib here.